### DISCUSSION TITLE
data is duplicating

### DISCUSSION THREAD

**Discussion by jishnuAs** [vitusortner/floor#737] [General]
data is duplicating but the primary key is autogenerate so that reason the primary key can't do anything the all data is coming from the backend

**Comment by dkaera** [comment]
@jishnuAs sorry, not clear
can you provide a sample of code that illustrates the use case

**Comment by dkaera** [comment]
@jishnuAs 
basically the primary key used to identify unique data. You must define the names of the columns for the primary key, which are responsible for the uniqueness of the data.
Here is SQLite [doc](https://www.sqlitetutorial.net/sqlite-primary-key/) on using a primary key 
and how-to use primary keys in [floor](https://pinchbv.github.io/floor/entities/#primary-keys)
